f600233d1b154d507b782f84604bcb12/update/sys_script_include_8a897895dbef9d14b1227ea5f3961968.xml

```diff
@@ -11,68 +11,77 @@ Payload is sent to the HOTH API for document generation.</description>
-							record.get(spec);
-							return record.name.toString();}) : null,
-						customSpecifications: evalPlan.custom_specifications ?evalPlan.custom_specifications.toString().split(",").map(function(spec) { 						
-							return spec;}) : null,
-						standardDifferentiators:  evalPlan.standard_differentiators ? evalPlan.standard_differentiators.toString().split(",").map(function(diff) { 
-							var record = new GlideRecord("x_g_dis_atat_eval_plan_differentiator");
-							record.get(diff);
-							return record.name.toString();}) : null,
-						customDifferentiators: evalPlan.custom_differentiators ? evalPlan.custom_differentiators.toString().split(",").map(function(diff) { 						
-							return diff;}) : null
-					}
-				};
-			}
-			else {
-				throw this.errUtil.createError("No Eval Plan found for this Acq Package. ", this.errUtil.INVALID_INPUT);
-			}
-		} catch(err) {
-			throw this.errUtil.createError("Error getting Eval Plan payload. " + err, this.errUtil.INVALID_INPUT);
-		}
-	},
+    getPayload: function() {
+        try {
+            var evalPlan = this.acquisitionPackage.evaluation_plan;
+            var projectOverview = this.acquisitionPackage.project_overview;
+            var customDifferentiatorsFlag = false; // if custom differentiators are specified
+            if (evalPlan) {
+                return {
+                    documentType: "EVALUATION_PLAN",
+                    templatePayload: {
+                        taskOrderTitle: projectOverview.title ? projectOverview.title.toString() : null,
+                        sourceSelection: evalPlan.source_selection ? evalPlan.source_selection.toString() : null,
+                        method: evalPlan.method ? evalPlan.method.toString() : null,
+                        standardSpecifications: evalPlan.standard_specifications ? evalPlan.standard_specifications.toString().split(",").map(function(spec) {
+                            var record = new GlideRecord("x_g_dis_atat_eval_plan_assessment_area");
+                            record.get(spec);
+                            return record.name.toString();
+                        }) : null,
+                        customSpecifications: evalPlan.custom_specifications ? evalPlan.custom_specifications.toString().split(",").map(function(spec) {
+                            return spec;
+                        }) : null,
+                        standardDifferentiators: evalPlan.standard_differentiators ? evalPlan.standard_differentiators.toString().split(",").filter(function(differentiatorSysId) {
+                            var record = new GlideRecord("x_g_dis_atat_eval_plan_differentiator");
+                            record.get(differentiatorSysId);
+                            customDifferentiatorsFlag = record.name.toString() === 'CustomDifferentiators'; // this standard differentiator is used to indicate the need include custom differentiators
```

Comment:
Note to self: Check whether this works when _CustomDifferentiators_ is not the last entry.

Reply:
Tested.  Definitely a problem.

Reply:
Fixed in https://github.com/dod-ccpo/atat-snow/pull/268/commits/29cf385bc025967b574300325f36941777f058b0
Placement of custom indicator in list has no impact.